submissions/vakulinina/js-post-oop/index.js

```diff
@@ -0,0 +1,62 @@
+class Inhabitant {
+   constructor(species, name, gender, legs, saying) {
+      this.species = species;
+      this.name = name;
+      this.gender = gender;
+      this.saying = saying;
+      this.legs = legs;
+      this.friends = [];
+   }
+
+   makeFriends(...friends) {
+      this.friends = [...friends];
+   }
+
+   getProperties() {
+      return [this.species, this.name, this.gender, this.legs, this.saying, this.getFriendsList(this.friends)];
+   }
+
+   getFriendsList(friends) {
+      return friends.length > 0 ? friends.map(friend => friend.name).join(', ') : 'no friends';
+   }
+
+   introduce() {
+      return this.getProperties().join('; ');
+   }
+}
+
+class Human extends Inhabitant {
+   constructor(name, gender, saying) {
+      super('human', name, gender, 2, saying);
+      this.hands = 2;
+   }
+
+   getProperties() {
+      let properties = super.getProperties();
+      properties.splice(3, 0, this.hands);
```

Comment:
This approach relies on an order of props and may easily become a bug when hierarchy grows.

Reply:
Note taken. Thank you!